sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -5,25 +5,146 @@ package com.azure.cosmos.implementation
 
 import com.azure.cosmos.CosmosClientBuilder
 import com.azure.cosmos.implementation.ImplementationBridgeHelpers.CosmosClientBuilderHelper
-import com.azure.cosmos.implementation.changefeed.implementation.ChangeFeedState
+import com.azure.cosmos.implementation.changefeed.implementation.{ChangeFeedState, ChangeFeedStateV1}
+import com.azure.cosmos.implementation.feedranges.{FeedRangeContinuation, FeedRangeEpkImpl, FeedRangeInternal}
+import com.azure.cosmos.implementation.query.CompositeContinuationToken
+import com.azure.cosmos.implementation.routing.Range
+import com.azure.cosmos.models.FeedRange
+import com.azure.cosmos.spark.{CosmosLoggingTrait, NormalizedRange}
 
-private[cosmos] object SparkBridgeImplementationInternal {
+// scalastyle:off underscore.import
+import scala.collection.JavaConverters._
+// scalastyle:on underscore.import
+
+private[cosmos] object SparkBridgeImplementationInternal extends CosmosLoggingTrait {
   def setMetadataCacheSnapshot(cosmosClientBuilder: CosmosClientBuilder,
                                metadataCache: CosmosClientMetadataCachesSnapshot): Unit = {
 
     val clientBuilderAccessor = CosmosClientBuilderHelper.getCosmosClientBuilderAccessor
     clientBuilderAccessor.setCosmosClientMetadataCachesSnapshot(cosmosClientBuilder, metadataCache)
   }
 
-  def extractLsnFromChangeFeedContinuation(continuation: String) : Long = {
+  def changeFeedContinuationToJson(continuation: String): String = {
+    ChangeFeedState
+      .fromString(continuation)
+      .toJson()
+  }
+
+  def extractLsnFromChangeFeedContinuation(continuation: String): Long = {
     val lsnToken = ChangeFeedState
       .fromString(continuation)
       .getContinuation
       .getCurrentContinuationToken
       .getToken
 
+    toLsn(lsnToken)
+  }
+
+  def overrideLsnInChangeFeedContinuation(continuation: String, newContinuationToken: String): String = {
+    val state = ChangeFeedState
+      .fromString(continuation)
+    val continuationToken = state.getContinuation.getCurrentContinuationToken
+    continuationToken.setToken(newContinuationToken)
+    state.toString
```

Comment:
the getCurrentContinuationToken will still ocntain it. BUt yes - the whole \" or not caused me quite some headache when going thorugh tests because even the backend does it inconsistently (in header with \" in the _lsn property without etc.

Reply:
But I am pretty optimistic that I found all permutation in the tests